### PR TITLE
added an error for variation using variation for parent in import and…

### PR DESCRIPTION
**Pull request by ghost**
… removed @access on 87

### All Submissions:

* [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Add an error so that a user importing a CSV doesn't have to stare at the loading bar for 3 hours wondering if there's a problem or not.

Closes # .

### How to test the changes in this Pull Request:

1. Repeat steps from [this issue](https://github.com/woocommerce/woocommerce/issues/21809)

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### Changelog entry

> Added an error to handle scenarios when a csv might have an existing SKU that is a variation and thus can't be used.

### CODE REVIEW

**Comment by ghettifish** [comment]
It might be nice to have some sort of error on the front end for the user as well. Right now this falls under the error message "A product with this ID already exists."

**Comment by ghettifish** [comment]
I should mention this is my first PR so be extra careful when looking over this one. This solved my problem, but it may not have been the best way to solve my problem.

File: includes/import/abstract-wc-product-importer.php [hunk 1]
```diff
@@ -434,6 +434,11 @@ protected function set_variation_data( &$variation, $data ) {
 			return new WP_Error( 'woocommerce_product_importer_missing_variation_parent_id', __( 'Variation cannot be imported: Missing parent ID or parent does not exist yet.', 'woocommerce' ), array( 'status' => 401 ) );
 		}
 
+		// Stop if parent is a product variation.
+		if ( $parent->post_type === 'product_variation' ) {
```
Comment (claudiosanches) [inline includes/import/abstract-wc-product-importer.php]: Better change to:

```php
if ( $parent->is_type( 'variation' ) ) {
```

File: includes/import/abstract-wc-product-importer.php [hunk 2]
```diff
@@ -434,6 +434,11 @@ protected function set_variation_data( &$variation, $data ) {
...
+			return new WP_Error( 'woocommerce_product_importer_parent_set_as_variation', __( 'Variation cannot be imported: Parent cannot be a product variation', 'woocommerce' ), array( 'status' => 401 ) );
```
Comment (claudiosanches) [inline includes/import/abstract-wc-product-importer.php]: Will probably should better like:

```
Variation cannot be imported: Parent product cannot be a variation.
```

**Review by claudiosanches — CHANGES_REQUESTED** [review]
Great work for your first Pull Request.
It's working, but I'm requesting a few changes just to follow our standards.

File: includes/import/abstract-wc-product-importer.php (hunk 2 above)
Comment (ghettifish) [reply includes/import/abstract-wc-product-importer.php]: Awesome! Thanks for the feedback. I just pushed those changes.

**Review by claudiosanches — APPROVED** [review]
Great job.
Thanks for your contribution.

**Pull request merged**